### ISSUE TITLE
Opening larger txt file slows atom / upon opening freeze for period of time

### ISSUE THREAD

**Issue by stebiger** [atom/atom#8836]
Hey,

I was just opening a larger text-file with atom editor on windows 7 and noticed the following reproducable behaviour:

Upon display of file atom editor freezed and did not respond to any actions.
After a period of time, 1-2 minutes, I was able to use the file.

The file had a size of 2.248kb - so we are not talking about a huge file.
This should in my humble opinion work smooth.

Also closing the file tab produced a short lag without responding to actions.

Regards,
Stefan

**Comment by mnquintana** [comment]
Is this file minified by any chance?

**Comment by stebiger** [comment]
No - it was just a captured output of my maven build

ie

mvn clean package > mvn.log

**Comment by mnquintana** [comment]
Could you post a gist with the contents of the file so we can try to debug this further? :mag:

**Comment by COLABORATI** [comment]
Opening this file with Atom causes the described behaviour:
http://newsatlas.io/scripts/d9294c98.scripts.js
yes, this is minified. You can unminify this with a Sublime Package called HTML-CSS-JS Prettify:
https://packagecontrol.io/packages/HTML-CSS-JS%20Prettify
(Did not find an unminify package for atom).

The unminified source, when opened in atom, does work ok, also scrolling is acceptable fast, but syntax highlighting does not work anymore. Geany shows this file without problems and highlights unminified source, too.

(Using Linux, Ubuntu 14.04).

**Comment by tolga9009** [comment]
I can confirm this. I have loaded up a ~6MB textfile (http://norvig.com/big.txt) and I was able to navigate in that file without any performance issues. After adding a some new lines somewhere in line 74500+, Atom freezes. This happens with the 32bit Windows stable release, aswell as the 1.4.0-beta0 release, running on Windows 8.1, Intel Xeon E3-1225v3, 8GB RAM.

One Core gets choked to 100%, the app freezes totally. After waiting some time (around 2 - 3 minutes), the app works again and I can continue my work.

**Comment by lee-dohm** [comment]
I just tried the following using Atom v1.5.0-dev-e16b17a on Mac OS X 10.11.2 (32GB of RAM though):

1. Open Atom in the atom/atom repository (so tree-view has something to crunch on)
1. Open the `big.txt` file from a local directory
1. Use Goto Line to go to line 74500
1. Press <kbd>Enter</kbd> several times

Observed no slow down or other issue. Atom shows as only using 79.5MB of memory.

@tolga9009 is it possible that your system is under a significant amount of memory pressure so Atom has to swap a lot?

**Comment by tolga9009** [comment]
Nope, actually, I'm running nothing else. Memory is about 1.8GB / 8GB, everything seems okay. The CPU is just choking on 1 Core.

Luckily, I have a Hackintosh Mac OS X 10.10.2 running on the same system, I will try it out here aswell (just latest stable and 1.4.0-beta0, since I don't have a build environment set up at the moment). I'll test it there and edit this post, to report back.

**Comment by lee-dohm** [comment]
@tolga9009 Most likely the freeze is the [whitespace package](https://github.com/atom/whitespace) removing trailing whitespace on save. One measurement on my machine took 7.6s for the `Whitespace.removeTrailingWhitespace` function to finish. When I disabled the whitespace package it only took 912ms.

**Comment by tolga9009** [comment]
I've tried disabling whitespace package on my Windows machine, it didn't fix the issue. However, I was able to identify the culprit: spell-check package. After disabling it, the freeze / lag is completely gone. The hang on saving `big.txt` is also noticeable under Windows and gets fixed by disabling whitespace package.

So, spell-check seems to block Atom completely during spell-checking. After the hang is over and Atom gets responsive again, when you try to make further edits, Atom freezes immediately again. So, spell-check package seems to be unusable for huge textfiles.

Interesting, that spell-checking doesn't lag under Mac OS X, but under Windows.

Cheers,
Tolga
